pkg/agent/integration_test.go

```diff
@@ -455,3 +457,106 @@ func Test_Non_Transparent_Proxy(t *testing.T) {
 		t.Fatalf("failed %v", err)
 	}
 }
+
+func stressCPU(load uint) []string {
+	return []string{
+		"xk6-disruptor-agent",
+		"stress",
+		"--duration",
+		"20s",
+		"--load",
+		fmt.Sprintf("%d", load),
+	}
+}
+
+const cpuLoadTolerance = 10
+
+func Test_CPUStressor(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct{
+		title string
+		load  uint
+	}{
+		{
+			title: "100% CPU",
+			load:  100,
+		},
+		{
+			title: "80% CPU",
+			load:  80,
+		},
+		{
+			title: "60% CPU",
+			load:  60,
+		},
+		{
+			title: "40% CPU",
+			load:  40,
+		},
+		{
+			title: "20% CPU",
+			load:  20,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.title, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.TODO()
+
+			// start the container with the agent using a fake upstream address because the agent requires it
+			gcr := testcontainers.GenericContainerRequest{
+				ProviderType: testcontainers.ProviderDocker,
+				ContainerRequest: testcontainers.ContainerRequest{
+					Image:      "ghcr.io/grafana/xk6-disruptor-agent",
+					Cmd:        stressCPU(tc.load),
+					Privileged: true,
+				},
+				Started: true,
+			}
+
+			agent, err := testcontainers.GenericContainer(ctx, gcr)
+			if err != nil {
+				t.Fatalf("failed to create agent container %v", err)
+			}
+			t.Cleanup(func() {
+				_ = agent.Terminate(ctx)
+			})
+
+			stats := []tcutils.ContainerStats{}
+			for {
+				state, err := agent.State(ctx)
+				if err != nil {
+					t.Fatalf("getting container status %v", err)
+				}
+
+				if !state.Running {
+					break
+				}
+
+				s, err := tcutils.Stats(context.TODO(), agent.GetContainerID())
+				if err != nil {
+					t.Fatalf("getting container stats %v", err)
+				}
+				stats = append(stats, s)
+			}
```

Comment:
I might be overthinking this, but I'm a bit scared of this loop potentially running too fast and skyrocketing the test memory usage. I think we could prevent this by simply adding a tiny sleep to "throttle" the sampling process:
```suggestion
				stats = append(stats, s)
				time.Sleep(170*time.Millisecond)
			}
```
Where 170 is sort of a weird number which is unlikely to be a multiple of the stressor slice (just in case). WDYT?

Reply:
An alternative way of computing the cpu average without storing samples is:
```go
iterations := 0
average := 0.0
for {
    //...
    average = (average * iterations + sample) / (iterations + 1)
    iterations++
}
```

Reply:
I just saw that tcutils.Start() includes a one second delay, which removes the need for delaying further in this loop. It might be worth mentioning it here in a comment though, just in case we get rid of it later and forget to add a delay here.

Reply:
I can add the comment, but the `Stats` function mentions it.

Reply:
> An alternative way of computing the cpu average without storing samples is:

Can change it. I was collecting them for debugging purposes.